core/scripts/create_adapter_plugins.py

```diff
@@ -0,0 +1,162 @@
+    parsed = parser.parse_args()
+
+    if parsed.title_case is None:
+        parsed.title_case = parsed.adapter.title()
+
+    if parsed.dependency:
+        #['a', 'b'] => "'a',\n        'b'"; ['a'] -> "'a',"
+        parsed.dependency = '\n        '.join(
+            "'{}',".format(d) for d in parsed.dependency
+        )
+    else:
+        parsed.dependency = '<INSERT DEPENDENCIES HERE>'
+
+    if parsed.email is not None:
+        parsed.email = "'{}'".format(parsed.email)
+    else:
+        parsed.email = '<INSERT EMAIL HERE>'
+    if parsed.author is not None:
+        parsed.author = "'{}'".format(parsed.author)
+    else:
+        parsed.author = '<INSERT AUTHOR HERE>'
+    if parsed.url is not None:
+        parsed.url = "'{}'".format(parsed.url)
+    else:
+        parsed.url = '<INSERT URL HERE>'
+    return parsed
+
+
+
+def main():
+    parsed = parse_args()
+    dest = pj(parsed.root, parsed.adapter)
+    if os.path.exists(dest):
+        raise Exception('path exists')
+
+    adapters_path = pj(dest, 'dbt', 'adapters', parsed.adapter)
+    include_path = pj(dest, 'dbt', 'include', parsed.adapter)
+    os.makedirs(adapters_path)
+    os.makedirs(pj(include_path, 'macros'))
+
+    # namespaces!
+    with open(pj(dest, 'dbt', '__init__.py'), 'w') as fp:
+        fp.write(NAMESPACE_INIT_TEMPLATE)
+    with open(pj(dest, 'dbt', 'adapters', '__init__.py'), 'w') as fp:
+        fp.write(NAMESPACE_INIT_TEMPLATE)
+    with open(pj(dest, 'dbt', 'include', '__init__.py'), 'w') as fp:
+        fp.write(NAMESPACE_INIT_TEMPLATE)
+
+    # setup script!
+    with open(pj(dest, 'setup.py'), 'w') as fp:
+        fp.write(SETUP_PY_TEMPLATE.format(adapter=parsed.adapter,
+                                          version=parsed.package_version,
+                                          author_name=parsed.author,
+                                          author_email=parsed.email,
+                                          url=parsed.url,
+                                          dbt_core_version=parsed.dbt_core_version,
+                                          dependencies=parsed.dependency))
+
+
+    # adapter stuff!
+    with open(pj(adapters_path, '__init__.py'), 'w') as fp:
+        fp.write(ADAPTER_INIT_TEMPLATE.format(adapter=parsed.adapter,
+                                              title_adapter=parsed.title_case))
+
+    # macro/project stuff!
+    with open(pj(include_path, '__init__.py'), 'w') as fp:
+        fp.write(INCLUDE_INIT_TEMPLATE)
+
+    with open(pj(include_path, 'dbt_project.yml'), 'w') as fp:
+        fp.write(PROJECT_TEMPLATE.format(adapter=parsed.adapter,
+                                         version=parsed.project_version))
+
+    # TODO:
+    # - bare class impls for mandatory subclasses
+    #       (ConnectionManager, Credentials, Adapter)
+    # - impls of mandatory abstract methods w/explicit NotImplementedErrors
+
+
+if __name__ == '__main__':
+    main()
```

Comment:
is this file still useful? seems like it was used to generate this branch, but should not make it into dev

Reply:
This is actually a script I wrote to help with generating new adapters, I was planning on using it for Presto and maybe shipping it with dbt-core so plugin authors can use it. It should be in that branch though, not this one, and it's not very finished! Good catch on this, I'll remove it.